addons/utm/models/utm_source.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import _, api, fields, models, tools
+
+
+class UtmSource(models.Model):
+    _name = 'utm.source'
+    _description = 'UTM Source'
+
+    name = fields.Char(string='Source Name', required=True)
+
+    _sql_constraints = [
+        ('unique_name', 'UNIQUE(name)', 'The name must be unique'),
+    ]
+
+    @api.model
+    def create(self, vals):
+        vals['name'] = self.env['utm.mixin']._generate_unique_name(self, vals.get('name'))
+        return super(UtmSource, self).create(vals)
+
+    def _generate_name(self, record, content):
+        """Generate the UTM source name based on the content of the source."""
+        if len(content) > 25:
+            content = f'{content[:25]}...'
+
+        create_date = record.create_date or fields.date.today()
+        create_date = fields.date.strftime(create_date, tools.DEFAULT_SERVER_DATE_FORMAT)
+        model_description = self.env['ir.model']._get(record._name).name
+        return _('%s (%s created on %s)', content, model_description, create_date)
```

Comment:
Would probably eithe rcall set unique, either ensure it is unique

Reply:
ok, so I remove the `create` overwrite as it's redundant